packages/integration-sdk-entity-validator/src/validator.ts

```diff
@@ -0,0 +1,66 @@
+import Ajv, { AnySchema, ErrorObject } from 'ajv';
+import addFormats from 'ajv-formats';
+import { addJ1Formats } from './j1Formats';
+
+export class EntityValidator {
+  private ajvInstance: Ajv;
+
+  constructor({ schemas }: { schemas?: AnySchema[] }) {
+    this.ajvInstance = addJ1Formats(
```

Comment:
I have a proposal for a different interface of this package, where this package provides the following functions:
- Fetch the _class schemas, e.g. `async fetchClassSchemas(): JsonSchemas[]`
- Fetch _type schemas of a certain integration, e.g. `async fetchTypeSchemas(integrationName: string): JsonSchemas[]`
- Run validation on an object (not `Entity`), with a passed in Ajv instance e.g.
```typescript
validateEntity(ajv: Ajv, obj: object): ValidationResult
```

This package **would not** have its own Ajv instance. That would instead be handled by the consumer, which allows them to configure the instance, and also handle its lifecycle (re-use it, or create a new one if they want).

What I'm proposing here does leak the detail of using Ajv to the consumer heavily, so you could say we're actually making an "Ajv utils" library not a generic validation library, but I think that's fine.

The reason for this drastic change of the API is that I don't think the validation logic can be abstracted away cleanly and promise the consumer that they don't need to know about how we actually run the validation.
Fetching the schemas is something the consumer will need to know about, especially fetching _type schemas, they might want to cache that or manage it in whatever way makes sense for them.
In addition, the results of the Ajv validator are not pleasant to work with. If we want to expose the Ajv validation result to the consumer, then we haven't really done anything useful to the consumer. They might as well just use Ajv by themselves without the need for this library.